Kickstarter-iOS/Features/PledgePaymentMethods/Controller/NoShippingPledgeRewardsSummaryViewController.swift

```diff
@@ -0,0 +1,314 @@
+      }
+  }
+
+  // MARK: - Configuration
+
+  func configureWith(
+    rewardsData: PostCampaignRewardsSummaryViewData,
+    bonusAmount: Double?,
+    pledgeData: PledgeSummaryViewData
+  ) {
+    self.viewModel.inputs
+      .configureWith(rewardsData: rewardsData, bonusAmount: bonusAmount, pledgeData: pledgeData)
+
+    self.view.setNeedsLayout()
+  }
+
+  // MARK: Styles
+
+  private func tableViewStyle(_ tableView: UITableView) {
+    tableView.separatorInset = .zero
+    tableView.contentInsetAdjustmentBehavior = .never
+    tableView.isScrollEnabled = false
+    tableView.delegate = self
+    tableView.rowHeight = UITableView.automaticDimension
+    tableView.separatorStyle = .none
+    tableView.backgroundColor = .ksr_white
+    tableView.translatesAutoresizingMaskIntoConstraints = false
+  }
+
+  private func rootStackViewStyle(_ stackView: UIStackView) {
+    stackView.axis = NSLayoutConstraint.Axis.vertical
+    stackView.spacing = Styles.grid(1)
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.translatesAutoresizingMaskIntoConstraints = false
+  }
+
+  private func separatorViewStyle(_ view: UIView) {
+    view.backgroundColor = .ksr_support_200
+    view.translatesAutoresizingMaskIntoConstraints = false
+  }
+
+  private func sectionHeaderViewStyle(_ view: UIView) {
+    view.translatesAutoresizingMaskIntoConstraints = false
+    view.frame = CGRectMake(0, 0, self.tableView.frame.size.width, self.tableView.frame.size.height)
+  }
+
+  private func sectionHeaderLabelStyle(_ label: UILabel) {
+    label.font = UIFont.ksr_subhead().bolded
+    label.textColor = UIColor.ksr_black
+    label.numberOfLines = 0
+    label.frame = CGRectMake(
+      CheckoutConstants.PledgeView.Inset.leftRight,
+      0,
+      self.tableView.frame.size.width,
+      PledgeRewardsSummaryStyles.Layout.sectionHeaderLabelHeight
+    )
+  }
+
+  private func tableViewContainerStyle(_ view: UIView) {
+    view.translatesAutoresizingMaskIntoConstraints = false
+    view.clipsToBounds = true
+  }
+
+  // MARK: - Helpers
+
+  private func setEntireViewToIsHidden(_ isHidden: Bool) {
+    self.view.isHidden = isHidden
+    self.pledgeTotalViewController.view.isHidden = isHidden
+    self.separatorView.isHidden = isHidden
+  }
+
+  // MARK: Apply DataSource Snapshot
+
+  private func applySnapshotToDataSource(from data: [PostCampaignRewardsSummaryItem]) {
+    var snapshot = NSDiffableDataSourceSnapshot<PledgeRewardsSummarySection, PledgeRewardsSummaryRow>()
+
+    /// Decipher header vs reward objects from the `[PostCampaignRewardsSummaryItem]` data object.
+    let headerItemData = data.compactMap { item -> PledgeExpandableHeaderRewardCellData? in
+      guard case let .header(data) = item else { return nil }
+      return data
+    }
+
+    let rewardData = data.compactMap { item -> PledgeExpandableHeaderRewardCellData? in
+      guard case let .reward(data) = item else { return nil }
+      return data
+    }
```

Comment:
This follows the pattern that the current data source class uses. I tried to focus only on implementing this new DiffableDataSource API and avoid changing how the table data is created and sent to this view controller from the view model.

I think there's an argument for this logic to live in the VM.